### ISSUE TITLE
Opening the app on linux fullscreens a full black screen

### ISSUE THREAD

**Issue by Blade04208** [mulaRahul/keyviz#280]
**Describe the bug**
Opening the app on linux fullscreens a full black screen

**Version**: v2.0.0-alpha2

**To Reproduce**
Open the program.

**Expected behavior**
Not having my screen be completely black!

**Screenshots**
[Screencast from 2024-06-07 22-32-47.webm](https://github.com/mulaRahul/keyviz/assets/74104258/492b3345-4646-477c-b70a-28ef2ec11ade)

**Desktop (please complete the following information):**

System Details Report
- **Date generated:**                              2024-06-07 22:37:04

Software Information:
- **Firmware Version:**                            P07CFL.028.240221.HQ
- **OS Name:**                                     EndeavourOS
- **OS Build:**                                    rolling
- **OS Type:**                                     64-bit
- **GNOME Version:**                               46
- **Windowing System:**                            Wayland
- **Kernel Version:**                              Linux 6.9.3-arch1-1
 

**Additional context**
here is the logs when opening via terminal
```
flutter: [ json:load ]: File: '/home/blade0/Documents/config.json' not found!
flutter: registered mouse listener
flutter: keyboard listener registered
flutter: [ json:load ]: File: '/home/blade0/Documents/style.json' not found!
[ERROR:flutter/runtime/dart_vm_initializer.cc(41)] Unhandled Exception: MissingPluginException(No implementation found for method setIgnoreMouseEvents on channel window_manager)
#0      MethodChannel._invokeMethod (package:flutter/src/services/platform_channel.dart:332)
<asynchronous suspension>
#1      WindowManager.setIgnoreMouseEvents (package:window_manager/src/window_manager.dart:699)
<asynchronous suspension>

[ERROR:flutter/runtime/dart_vm_initializer.cc(41)] Unhandled Exception: MissingPluginException(No implementation found for method setHasShadow on channel window_manager)
#0      MethodChannel._invokeMethod (package:flutter/src/services/platform_channel.dart:332)
<asynchronous suspension>
#1      WindowManager.setHasShadow (package:window_manager/src/window_manager.dart:667)
<asynchronous suspension>


(keyviz:4337): Gtk-CRITICAL **: 22:33:01.054: gtk_widget_get_scale_factor: assertion 'GTK_IS_WIDGET (widget)' failed
Segmentation fault (core dumped)

```
uhh good luck
linux is like 10x more varied than windows or macos can be so expect random-ass bugs (like this one) that only affect like 1 or 2 people

**Comment by mulaRahul** [comment]
I've seen this error before, about window_manager not being implemented on some flavours of Linux

**Comment by SJFCS** [comment]
I also encountered the same problem on archlinux. Is there any solution?

**Comment by BlastboomStrice** [comment]
Same on Debian Testing (trixie). I can turn it on and off, then I go to the black screen where the settings appear, but the app doesn't seem to work.

**Comment by pietro-lopes** [comment]
I'm having the same problem

OS: Pop!_OS 22.04 LTS x86_64
Kernel: 6.8.0-76060800daily20240311-generic
Shell: zsh 5.8.1
Resolution: 3840x2160
DE: GNOME 42.9
WM: Mutter
WM Theme: Pop
Theme: Pop-dark [GTK2/3]
Icons: Pop [GTK2/3]
CPU: Intel i5-7600 (4) @ 4.100GHz
GPU: NVIDIA GeForce GTX 1660 SUPER

**Comment by witt-bit** [comment]
My operating system information
```shell
 ██████████████████  ████████     witt@witt-cool
 ██████████████████  ████████     OS: Manjaro 24.2.0 Yonada
 ██████████████████  ████████     Kernel: x86_64 Linux 6.11.10-2-MANJARO
 ██████████████████  ████████     Uptime: 2d 5h 57m
 ████████            ████████     Packages: 1363
 ████████  ████████  ████████     Shell: zsh 5.9
 ████████  ████████  ████████     Resolution: 3840x1080
 ████████  ████████  ████████     DE: KDE
 ████████  ████████  ████████     WM: KWin
 ████████  ████████  ████████     GTK Theme: WhiteSur-Dark-solid-alt-blue [GTK2/3]
 ████████  ████████  ████████     Icon Theme: Tela-circle-manjaro-light
 ████████  ████████  ████████     Disk: 186G / 459G (43%)
 ████████  ████████  ████████     CPU: AMD Ryzen 5 4500U with Radeon Graphics @ 6x 2.375GHz
 ████████  ████████  ████████     GPU: AMD Radeon Graphics (radeonsi, renoir, LLVM 18.1.8, DRM 3.59, 6.11.10-2-MANJARO)
                                  RAM: -
```

After running the `./keyviz` command, the terminal has the following error
```shell
** (keyviz:996188): CRITICAL **: 15:22:41.281: Failed to read XDG desktop portal settings: GDBus.Error:org.freedesktop.portal.Error.NotFound: 未找到请求的设置

** (keyviz:996188): CRITICAL **: 15:22:41.285: Failed to read XDG desktop portal settings: GDBus.Error:org.freedesktop.portal.Error.NotFound: 未找到请求的设置

** (keyviz:996188): CRITICAL **: 15:22:41.285: Failed to read XDG desktop portal settings: GDBus.Error:org.freedesktop.portal.Error.NotFound: 未找到请求的设置

** (keyviz:996188): CRITICAL **: 15:22:41.287: Failed to read XDG desktop portal settings: GDBus.Error:org.freedesktop.portal.Error.NotFound: 未找到请求的设置

** (keyviz:996188): CRITICAL **: 15:22:41.288: Failed to read XDG desktop portal settings: GDBus.Error:org.freedesktop.portal.Error.NotFound: 未找到请求的设置
flutter: [ json:load ]: File: '/home/witt/Documents/config.json' not found!
flutter: registered mouse listener
flutter: keyboard listener registered
flutter: [ json:load ]: File: '/home/witt/Documents/style.json' not found!
[ERROR:flutter/runtime/dart_vm_initializer.cc(41)] Unhandled Exception: MissingPluginException(No implementation found for method setIgnoreMouseEvents on channel window_manager)
#0      MethodChannel._invokeMethod (package:flutter/src/services/platform_channel.dart:332)
<asynchronous suspension>
#1      WindowManager.setIgnoreMouseEvents (package:window_manager/src/window_manager.dart:699)
<asynchronous suspension>

[ERROR:flutter/runtime/dart_vm_initializer.cc(41)] Unhandled Exception: MissingPluginException(No implementation found for method setHasShadow on channel window_manager)
#0      MethodChannel._invokeMethod (package:flutter/src/services/platform_channel.dart:332)
<asynchronous suspension>
#1      WindowManager.setHasShadow (package:window_manager/src/window_manager.dart:667)
<asynchronous suspension>

flutter: ⬇️ [Left Click] not hotkey, returning...
flutter: ⬇️ [Drag] not hotkey, returning...
flutter: ⬇️ [Left Click] not hotkey, returning...
flutter: ⬇️ [Left Click] not hotkey, returning...
flutter: ⬇️ [Drag] not hotkey, returning...
flutter: ⬇️ [Left Click] not hotkey, returning...
flutter: keyboardEvents: {2253351: {8589934848: ⬇️[control](🟢)}}
flutter: ⬇️ [Control Left]
flutter: ⬆️ [Control Left]
flutter: ⬇️ [Left Click] not hotkey, returning
```

In addition, the interface is completely black, but the pressed key will be displayed, as shown in the figure
![image](https://github.com/user-attachments/assets/2100715d-a0d8-4ca1-9dd9-7225cd8ff6bb)


I hope the above information can help you

**Comment by aum7** [comment]
same here
linux mint 21.3
- dl & unzip the code
- run by clicking on icon inside folder
- screen goes black, mouse circle ...
- keypress is not detected / displayed
- mouse circle follows mouse cursor

![keyviz black screen](https://github.com/user-attachments/assets/1cf17df9-7eb0-47ac-919b-c59b215b4299)

**Comment by hayyaun** [comment]
Any solution for manjaro?

**Comment by KaKi87** [comment]
Hi @mulaRahul,
Any news on this please ? 🙏 
Thank you

**Comment by wurenny** [comment]
same black screen in LinuxMint 22.1(Xia)

**Comment by conventoangelo** [comment]
Hi, I'm currently working on porting a Flutter app to Linux and have encountered the same issue. The problem arises because two methods from the `window_manager` package (a dependency of keyviz), namely `hasShadow` and `setIgnoreMouseEvents`, are called. However, these methods are not yet properly implemented on Linux. Both currently only work on Windows and macOS.

To address this, either:
1. Keyviz switches and use a different package (though this would require rewriting parts of the app).
2. Someone contributes to the `window_manager` package repository to implement the missing functionality.
3. Maintainer @mulaRahul sells their soul and go above and beyond to make an original window management implementation/solution for cross-platform desktops

TL;DR: It seems unlikely this will be resolved (in the near future?)

References:
- [`hasShadow` documentation](https://pub.dev/documentation/window_manager/latest/window_manager/WindowManager/hasShadow.html)
- https://github.com/leanflutter/window_manager/issues/89#issuecomment-1064716882

**Comment by AlexRslDev** [comment]
Same here ✋, Ubuntu 24.04.2. Anyone found a solution for it?

**Comment by kuqilin** [comment]
Same here.
OS: KUbuntu 24.04

For the journal:

```bash
kuqilin@UTG:~/Downloads/keyviz-v2.0.0a2-linux$ ./keyviz 
flutter: registered mouse listener
flutter: keyboard listener registered
[ERROR:flutter/runtime/dart_vm_initializer.cc(41)] Unhandled Exception: MissingPluginException(No implementation found for method setIgnoreMouseEvents on channel window_manager)
#0      MethodChannel._invokeMethod (package:flutter/src/services/platform_channel.dart:332)
<asynchronous suspension>
#1      WindowManager.setIgnoreMouseEvents (package:window_manager/src/window_manager.dart:699)
<asynchronous suspension>

[ERROR:flutter/runtime/dart_vm_initializer.cc(41)] Unhandled Exception: MissingPluginException(No implementation found for method setHasShadow on channel window_manager)
#0      MethodChannel._invokeMethod (package:flutter/src/services/platform_channel.dart:332)
<asynchronous suspension>
#1      WindowManager.setHasShadow (package:window_manager/src/window_manager.dart:667)
<asynchronous suspension>


```
